api/v1alpha1/envoyproxy_types.go

```diff
@@ -213,6 +222,14 @@ type EnvoyProxyList struct {
 	Items           []EnvoyProxy `json:"items"`
 }
 
+// OverloadManager defines the configuration for the Overload Manager in envoy.
+type OverloadManager struct {
+	// The appropriate number of bytes can be different from system to system.
```

Comment:
we shouldnt add this API field if we cannot suggest a prescriptive way for a user to come up with a value to insert

Reply:
The google VRP example uses a [2GiB max heap size](https://www.envoyproxy.io/docs/envoy/latest/configuration/best_practices/edge) when [executing a docker container](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/google_vrp#execution-environment) with 3GB of memory.

Proposal:
- By default, set MaxHeapSize to 70% of the envoy container's memory limit (which is known to EG, since it's generating the PodTemplate). If the limit is not set - don't configure anything. 
- Allow users to override this setting - configuring a custom percent. 

@arkodg , @tmsnan  - WDYT?

Reply:
hey @guydc agree with the approach ! since EG creates the envoy proxy, we have the the exact memory specs for the envoy proxy fleet
 imo we'll need to gather some more data before we decide on the the max heap and threshold values

Reply:
chatted with @KBaichoo who advised that we could consider a tighter limit like 80% to optimize resources, thanks !